disperser/dataapi/ejector.go

```diff
@@ -0,0 +1,124 @@
+package dataapi
+
+import (
+	"context"
+	"sort"
+	"sync"
+
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+)
+
+// The caller should ensure "stakeShare" is in range (0, 1].
+func stakeShareToSLA(stakeShare float64) float64 {
+	switch {
+	case stakeShare > 0.1:
+		return 0.975
+	case stakeShare > 0.05:
+		return 0.95
+	default:
+		return 0.9
+	}
+}
+
+// operatorPerfScore scores an operator based on its stake share and nonsigning rate. The
+// performance score will be in range [0, 1], with higher score indicating better performance.
+func operatorPerfScore(stakeShare float64, nonsigningRate float64) float64 {
+	if nonsigningRate == 0 {
+		return 1.0
+	}
+	sla := stakeShareToSLA(stakeShare)
+	perf := (1 - sla) / nonsigningRate
+	return perf / (1.0 + perf)
+}
+
+func computePerfScore(metric *OperatorNonsigningPercentageMetrics) float64 {
+	return operatorPerfScore(metric.StakePercentage, metric.Percentage/100.0)
+}
+
+type ejector struct {
+	logger     logging.Logger
+	transactor core.Transactor
+	metrics    *Metrics
+
+	// For serializing the ejection requests.
+	mu sync.Mutex
+}
+
+func newEjector(logger logging.Logger, tx core.Transactor, metrics *Metrics) *ejector {
+	return &ejector{
+		logger:     logger.With("component", "Ejector"),
+		transactor: tx,
+		metrics:    metrics,
+	}
+}
+
+func (e *ejector) eject(ctx context.Context, nonsigningRate *OperatorsNonsigningPercentage, mode string) error {
```

Comment:
`mode` isn't used

Reply:
Actually this isn't needed even after fixing TODO -- removed it!